### ISSUE TITLE
Boost filesystem not declared

### ISSUE THREAD

**Issue by DalInar** [aeantipov/opendf#5]
Compilation fails with the following message:

```
[ 73%] Building CXX object prog/CMakeFiles/hub_df_cubic1d.dir/hub_df.cpp.o
/u/home/oryx/ALPSCore_Joe/opendf/prog/hub_df.cpp: In function ‘void run(alps::params_ns::params)’:
/u/home/oryx/ALPSCore_Joe/opendf/prog/hub_df.cpp:134:30: error: ‘boost::filesystem’ has not been declared
         bool resume = boost::filesystem::exists(output_file);
```

I have the boost filesystem library installed, though I'm not sure how opendf is supposed to learn about it (is it via ALPSCore? I don't think boost::filesystem is an ALPSCore requirement.)  

I was seeing issues with the filesystem library when installing Maxent, which I learned was fixed by removing the dependency:
https://github.com/CQMP/Maxent/commit/0defb322e297e08f6a62027be05953bfc55d7c82

**Comment by jpfleblanc** [comment]
Can you show your cmake output?

Also, whatever script you used to compile.  I will send you one that I
recently used.  Where are you doing this? cori?

-james

**Comment by DalInar** [comment]
I'm compiling on Shakuras. Here's the cmake input/output:
```
[oryx@shakuras 000build]$ cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=/home/oryx/ALPSCore_Joe/install/ -DEIGEN3_INCLUDE_DIR=/home/oryx/software/install/include/eigen3/ -DFFTW_ROOT=/home/oryx/software/install/ .. | tee cmake_output

opendf configuration

-- BUILD_DIR: /u/home/oryx/ALPSCore_Joe/opendf/000build
-- SOURCE_DIR: /u/home/oryx/ALPSCore_Joe/opendf
-- Build type: Release
-- ALPSCore includes:
-- ALPSCore libraries: alps::alps-hdf5;alps::alps-accumulators;alps::alps-mc;alps::alps-params
-- Eigen3 includes: /home/oryx/software/install/include/eigen3/
-- Fetching gftools
-- GFTools includes : /u/home/oryx/ALPSCore_Joe/opendf/000build/gftools-prefix/src/gftools
-- Found FFTW: /home/oryx/software/install/lib/libfftw3.so
-- FFTW includes: /home/oryx/software/install/include
-- Lattices : cubic1d;cubic2d;cubic3d;cubic4d;triangular;square_nnn
-- DF: cubic1d
-- DF: cubic2d
-- DF: cubic3d
-- DF: cubic4d
-- DF: triangular
-- DF: square_nnn
-- Configuring done
-- Generating done
-- Build files have been written to: /u/home/oryx/ALPSCore_Joe/opendf/000build

```

**Comment by jpfleblanc** [comment]
Yup, I'd say alpscore doesn't have boost properly:

I compile opendf with a single line

CC=mpicc CXX=mpicxx  cmake
-DALPSCore_DIR=/home/jpfleblanc/alps_core/ALPSCore-0.5.4/install/share/ALPSCore/
-DCMAKE_INSTALL_PREFIX=/home/jpfleblanc/alps_core/opendf/install/
-DCMAKE_BUILD_TYPE=RELEASE  ..

It's unclear to me if I've compiled opendf against newer alpscore versions.
I'll look and see, but I don't think so.  If alex removed boost components
from alpscore this could be the reason.

**Comment by aeantipov** [comment]
Joe, should be fixed now with 74f90de. git pull and reinstall

**Comment by DalInar** [comment]
Awesome sauce, that seems to have fixed the issue.  Thanks Andrey!
